File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int n = A.length;
        int[] result = new int[n];
        int[] frequency = new int[n + 1];
        int commonCount = 0;

        for (int i = 0; i < n; i++) {
            // Increment frequency for element in A
            if (++frequency[A[i]] == 2) {
                commonCount++;
            }
            // Increment frequency for element in B
            if (++frequency[B[i]] == 2) {
                commonCount++;
            }
            // Store current prefix count
            result[i] = commonCount;
        }

        return result;
    }
}

File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.js
var findThePrefixCommonArray = function (A, B) {
    const n = A.length;
    const result = new Array(n).fill(0);
    const frequency = new Array(n + 1).fill(0);
    let commonCount = 0;

    for (let i = 0; i < n; i++) {
        if (++frequency[A[i]] === 2) {
            commonCount++;
        }
        if (++frequency[B[i]] === 2) {
            commonCount++;
        }
        result[i] = commonCount;
    }

    return result;
};